Fethas Carnival of Horrors part 4: I am in your greytide guarding ur newbs

Aurorablade opened the pull request:
Ports guardians from TG (i cannot for the life of me find the orignal PR for this....)

Guardians are entities such as spirits that inhabit a person and help them out.

They can manifest or recall from a person, but if they get 10 meters away they are pulled back (need to fix this in the code so they demanifest...)
They are technically invincible but any damage they DO take is transferred to their user, and if the user dies, they die.

Currently commented out:dusting the user when they die...i can put it back in, but i am unsure of it so commented out.

Guardian Themes:
Magical(Stand/Spirit): Summoned from a deck of tarot cards. CURRENTLY NOT IN GAME! I WOULD LOVE SUGGESTIONS ON HOW THIS SHOULD BE SPAWNED AND BY WHAT!

Tech(Holoparasite):Summoned by hypo injection. Bought from traitor uplink.

Bio(Scarab swarm, currently represented by a worm thing):Summoned by egg cluster. CURRENTLY NOT IN GAME! I WOULD LOVE SUGGESTIONS ON HOW THIS SHOULD BE SPAWNED AND BY WHAT!

Types of Guardians:

Chaos: As a Chaos type, you have only light damage resistance, but will ignite any enemy you bump into. In addition, your melee attacks will randomly teleport enemies.
Ignites mobs on contact. Teleports enemies away. Will not ignite user.

Standard:As a standard type you have no special abilities, but have a high damage resistance and a powerful attack capable of smashing through walls.
You punch things..alot of things..you say a battlecry when you punch things (getready for HONKHONKHONKHONK)

Support:
As a Support type, you may toggle your basic attacks to a healing mode. In addition, Alt-Clicking on an adjacent mob will warp them to your bluespace beacon after a short delay.

Two modes, healer and combat. combat is low damage. Healing mode will heal a thing on attack by 5.
Can set up a beacon and warp mobs to it.

Ranged:
As a ranged type, you have only light damage resistance, but are capable of spraying shards of crystal at incredibly high speed. You can also deploy surveillance snares to monitor enemy movement. Finally, you can switch to scout mode, in which you can't attack, but can move without limit.

BOMB:
As an explosive type, you have only moderate close combat abilities, but are capable of converting any adjacent item into a disguised bomb via alt click.

These are chosen at random on someone using the creator item. There are choosable variants that are admin spawnable.

ZomgPonies commented:
Interesting for events, seems self-contained, :+1:

Mijhead commented:
![yup _efd9e8dcac01dd8433fca5412ac559c8](https://cloud.githubusercontent.com/assets/12399664/11305796/a24f3a0e-8fae-11e5-8844-de304aeda565.jpg)

mrsdobbins commented:
http://i.imgur.com/fui2fTi.gif
SEKAI ICHI.

Spacemanspark commented:
:+1:

Ultimate-Chimera commented:
:+1:

Fox-McCloud commented:
I was going to port these, because they're pretty neat, but never got around to it.

Should be interesting.

taukausanake commented:
I thought people didn't like this because it was OP? I really like this but have only one feature I'd like to add. For a Revenant event give a random chance for a Guardian to be spawned/granted/adminbused and give the Guardian the objective to kill the Revenant. Spirit against spirit. It would give the chaplain a break for a while

Aurorablade commented:
eh its an odd issue on the opness, the solution there was to have the summoner be dusted when they died (IE Uncloneable)...i am unsure on that but will keep it in if its the only way (IE raise a TC cost) i also would like feedback on where the Bio and Terot ones can be gained.

Fox-McCloud commented on code/modules/mob/mob_helpers.dm:
```diff
@@ -142,7 +142,12 @@ proc/isembryo(A)
 		return 1
 	return 0
 
-proc/isobserver(A)
```
github, why you do this.

Aurorablade replied:
OH YOU ADDED SOMETHING BETWEEN TWO LINES?! HERE LET ME JUST SHOW YOU DELETED THE WHOLE THING!

Tastyfish replied:
The line's different from the original one, you silly people.

Fox-McCloud commented:
>These are chosen at random on someone using the creator item. There are choosable variants that are admin spawnable.

This is incorrect; the one from the traitor uplink allows you to select which one it will be when you inject it (thus why it's a subtype called pick with the random var set to false).

IMO, a good thing.

Aurorablade commented:
oh i missed that...

Mijhead commented:
<BlackCat> I feel like Stand users should spawn in vampire rounds.
<Mrs_Dobbins> Nah
<Mrs_Dobbins> Actually thats a cool idea
<Mrs_Dobbins> b-but they'd need some objective related to the vampires
<BlackCat> Though that'd probably require Vampires to get better or also have stands.
<Mrs_Dobbins> Just make the vampires more like the pillar men

;^)

mrsdobbins commented:
B-but shadowlings are, to all intents and purposes just pillar men...

Aurorablade commented:
@mijhead ...That gives me an idea..but i would need to limit it to like one or two types (We do NOT need vampires running around with Bomb and choas sprits) and i still need to figure out where to spawn Bio and Magic...Magic COULD go to wizard...MAYBE go to wizard....

Mijhead commented:
What about as an evolution of some sorts of borers?

Dumbdumn5 commented:
:+1: X :100:
Now I can finally have a way to say that I'm somebody's guardian angel without being a small card or a seal that does nothing but make rude comments.

Earthdivine commented:
Recommend giving Magic to the Wizard, he could stand having a second man to watch over him for Scientists with sprays.

Aurorablade commented:
yeah will do that. Also for vampires..i was considering a bat, as a power, limited to healer,standard and ranged/scout type. What i need to figure out is..make the power unuseable while a bat is bound to you...and thinking of if it would overlap with the Summon Bat power....

.....Also maybe i should make a admin only seal version for @Dumbdumn5

Aurorablade commented:
not gonna do the vampire thing..AM going to do the seal thing, i need someone to tell me how to make it truely adminbussable here so it dosen't pick a random player for it

Aurorablade commented:
i also just realized i forgot to give the magic creator to the wizard in last commit...i knew i was forgetting something...

Aurorablade commented:
Changed Seal on Feedback, Fixed Intent so it stops nuzzling things..hopefully....
Added wizard terot to spellbook

Help needed on:
Admin Seal needs to not pick a random player, but the admin who uses it and attach to a player.
Where do i put the Bio themed ones?

Aurorablade commented:
Added bio guardian spawner to Xenden asteroid treasure
modified battlecry, I need someone to CHECK the code here mainly cuase its not triggering when i a spawn a mob to punch, though that was before i did one change.

LAST THING TO DO:
Make the seal not choose a random player but the admin attempting to use it...no IDEA WHAT I AM DOING.

Earthdivine commented:
Throwing this here since I said it in the super busy chat, you probably will have to have it check for players with R_ADMIN or something. Forgot what it is.

Aurorablade commented:
Tried a swarmer fic seems to work (when i move away from what i dismantle..it does stop the process)
Adminbuss seal needs a double check on the code.
Otherwise done.
Review plz

Aurorablade commented:
well i mean review when i can commit this....*watchs git for windows

TheDZD commented on code/game/asteroid.dm:
```diff
@@ -97,7 +97,7 @@ var/global/max_secret_rooms = 6
 			walltypes = list(/turf/simulated/mineral/random/high_chance=1)
 			floortypes = list(/turf/simulated/floor/plating/airless/asteroid, /turf/simulated/floor/beach/sand)
-			treasureitems = list(/obj/item/clothing/mask/facehugger=1,/obj/item/stack/sheet/animalhide/xeno=2,/obj/item/clothing/suit/xenos=2,/obj/item/clothing/head/xenos=2)
+			treasureitems = list(/obj/item/clothing/mask/facehugger=1,/obj/item/stack/sheet/animalhide/xeno=2,/obj/item/clothing/suit/xenos=2,/obj/item/clothing/head/xenos=2,/obj/item/weapon/guardiancreator/biological=4)
```
The chance for a fairly powerful item probably shouldn't be that high, might be best to set its weight to 1.

TheDZD commented:
Besides that, I didn't find much to point out. Looks good otherwise.

Aurorablade commented:
okay commiting the treasure room tweak BUT can anyone see a reason WHY the battle cry is not triggering on attacking a mob?

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+
+/obj/item/projectile/guardian
+	name = "crystal spray"
+	icon_state = "guardian"
```
You never included this icon; it'll be invisible.

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	minbodytemp = 0
+	maxbodytemp = 10000000
+	min_oxy = 0
+	max_oxy = INFINITY
```
Shouldn't this be 0?

tigercat2000 replied:
If it was 0, the guardians would take damage any time they breath in oxygen.

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	min_oxy = 0
+	max_oxy = INFINITY
+	min_tox = 0
+	max_tox = INFINITY
```
as above

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	min_tox = 0
+	max_tox = INFINITY
+	min_co2 = 0
+	max_co2 = INFINITY
```
as above

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	min_co2 = 0
+	max_co2 = INFINITY
+	min_n2  = 0
+	max_n2  = INFINITY
```
as above

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	min_n2  = 0
+	max_n2  = INFINITY
+	attacktext = "punches"
+	maxHealth = 100000 //The spirit itself is invincible
```
Should be infinity.

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	max_n2  = INFINITY
+	attacktext = "punches"
+	maxHealth = 100000 //The spirit itself is invincible
+	health = 100000
```
Should be infinity.

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	var/magic_fluff_string = " You draw the Coder, symbolizing bugs and errors. This shouldn't happen! Submit a bug report!"
+	var/tech_fluff_string = "BOOT SEQUENCE COMPLETE. ERROR MODULE LOADED. THIS SHOULDN'T HAPPEN. Submit a bug report!"
+	var/bio_fluff_string = "Your scarabs fail to mutate. This shouldn't happen! Submit a bug report!"
+	var/admin_fluff_string = "URK URF!"//the wheels on the bus...
```
Should probably add the butcher result of this as being ectoplasm, as well; though it's by no means high priority.

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+			//		qdel(W)
+			//summoner.dust()
+			//ghostize()
+			//qdel(src)
```
I realize you wanted to get rid of the um dust() thing, but shouldn't it still delete the parasite? Just checking.

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+/mob/living/simple_animal/hostile/guardian/gib()
+	if(summoner)
+		summoner << "<span class='danger'><B>Your [src] was blown up!</span></B>"
+		summoner.gib()
```
Just wanted to make sure you're ok with including this; you didn't want Dust() included--is this ok?

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	ghostize()
+	qdel(src)
+
+/mob/living/simple_animal/hostile/guardian/start_pulling(var/atom/movable/AM)
```
Don't see the big deal about them being able to pull things.

Fox-McCloud commented on code/modules/mob/living/simple_animal/guardian/guardian.dm:
```diff
@@ -0,0 +1,895 @@
+	G << "You are capable of manifesting or recalling to your master with verbs in the Guardian tab. You will also find a verb to communicate with them privately there."
+	G << "While personally invincible, you will die if [user.real_name] does, and any damage dealt to you will have a portion passed on to them as you feed upon them to sustain yourself."
+	G << "[G.playstyle_string]"
+	user.verbs += /mob/living/proc/guardian_comm
```
forgot guardian recall and guardian reset
